openedx/features/idea/views.py

```diff
@@ -27,11 +33,27 @@ class IdeaListingView(ListView):
     template_engine = 'mako'
 
 
-class IdeaCreateView(View):
+@method_decorator(login_required, name='dispatch')
```

Comment:
How about using Mixin for class based views ?
from django.contrib.auth.mixins import LoginRequiredMixin